sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/subscribe/SubscriptionCheckboxFactory.groovy

```diff
@@ -0,0 +1,65 @@
+package life.qbic.portal.sampletracking.components.projectoverview.subscribe
+
+import com.vaadin.ui.CheckBox
+import groovy.util.logging.Log4j2
+import life.qbic.business.project.subscribe.Subscriber
+import life.qbic.portal.sampletracking.Constants
+import life.qbic.portal.sampletracking.communication.notification.NotificationService
+import life.qbic.portal.sampletracking.components.projectoverview.ProjectSummary
+
+/**
+ * <b>Checkboxes to handle project subscription</b>
+ *
+ * <p>Creates a checkbox for a {@link ProjectSummary} and adds a listener that triggers unsubscription from the project</p>
+ *
+ * @since 1.0.0
+ */
+@Log4j2
+class SubscriptionCheckboxFactory {
```

Comment:
You can move the map in here and divert the responsibility of delivering the right object to the factory (out of the map)

Reply:
but in my understanding the factory does create an object rather than containing it

Reply:
This is a detail the factory should decide. There is no need for any of the components outside of it to know where the element came from